### ISSUE TITLE
Optimize fill_holes, skip RegionProperties

### ISSUE THREAD

**Issue by dchaley** [dchaley/deepcell-imaging#357]
Part of: #322 

After #350 we gathered the data shown in #322 , which showed a lot of time spent in fill_holes.

The [`fill_holes` function](https://github.com/vanvalenlab/deepcell-toolbox/blob/e8c1277ee4243bc6a34916d554d0c2eab0cf7505/deepcell_toolbox/utils.py) calls the scikit [`regionprops` function](https://github.com/scikit-image/scikit-image/blob/e3e7c48b76d38a5f92db6ff9f355f933f9f977ff/skimage/measure/_regionprops.py), which does a lot of work creating objects. It also has some indirection around `setattr` which appears strictly unnecessary in our situation.

We only use a handful of properties from the region-properties objects as seen in the fill_holes code:

```python
    props = regionprops(np.squeeze(label_img.astype('int')), cache=False)
    for prop in props:
        if prop.euler_number < 1:

            patch = output_image[prop.slice]

            filled = remove_small_holes(
                ar=(patch == prop.label),
                area_threshold=size,
                connectivity=connectivity)

            output_image[prop.slice] = np.where(filled, prop.label, patch)
```

* prop.euler_number
* prop.slice
* prop.label

Of these, slice and label are returned by the [`findobjects` function](https://github.com/scikit-image/scikit-image/blob/e3e7c48b76d38a5f92db6ff9f355f933f9f977ff/skimage/measure/_regionprops.py#L1383-L1402) called by `regionprops` :

```python
    objects = ndi.find_objects(label_image)
    for i, sl in enumerate(objects):
        if sl is None:
            continue

        label = i + 1

        props = RegionProperties(
            sl,
            label,
            label_image,
            intensity_image,
            cache,
            spacing=spacing,
            extra_properties=extra_properties,
            offset=offset_arr,
        )
        regions.append(props)
```

And `euler_number` only uses the image & number of dimensions:

```python
    @property
    def euler_number(self):
        if self._ndim not in [2, 3]:
            raise NotImplementedError(
                'Euler number is implemented for ' '2D or 3D images only'
            )
        return euler_number(self.image, self._ndim)
```

the image being the slice from the main image:

```python
    @property
    @_cached
    def image(self):
        return self._label_image[self.slice] == self.label
```

There's an opportunity here to skip all of region properties. According to this:

![Image](https://github.com/user-attachments/assets/074c2c59-c1c9-4ce2-9471-7ba174ccda45)

we're spending as much of 20% of `fill_holes` building these objects.

Nonetheless the `euler_number` operation itself remains the bulk of this code; it's spending a lot of time allocating buffers for padded image data. That's another story for another day…
